packages/plugin-tooltip/src/utility/input.ts

```diff
@@ -66,6 +70,7 @@ export const updateLinkView: Updater = (view, $) => {
     const { firstChild, lastElementChild } = $;
     if (!(firstChild instanceof HTMLInputElement) || !(lastElementChild instanceof HTMLButtonElement)) return;
 
+    firstChild.focus();
```

Comment:
This line causes a regression that input view on mobile cannot be closed. I have to remove this line.

Reply:
@Saul-Mirone thanks for the heads up--I hadn't tested on mobile yet, but I just did and I'm in fact unable to create links on iOS 15.1 with the current demo. Are you able to do so?

Reply:
I take it back--I am able to create links on iOS on the current demo (that was user error)!

Reply:
The link creation had a bug before and I just fixed it yesterday. That's the time I found this regression.

Reply:
@Saul-Mirone would you feel comfortable calling focus on the input if we query for mobile device or not? It's kind of tedious inserting an image or link now--I'd like to make that as fluid as possible. 

Is there another approach altogether that would expedite this behavior? In an ideal world, clicking the image icon in the menu and/or selecting the image option from the slash menu would open a modal that lets the user specify the src/href, and the input would already be focused. I realize this seems difficult with the extant action options. 

Is there a way to create a prosemirror action that prompts the user for input mid-action, then does something with that input (if provided)?

Reply:
I also found that this code also leads to strange behavior on desktop. The input cannot be hide by the first click. I'll investigate a little to see the way to implement this.

Reply:
Yesterday I wrote a fairly simple modal and custom action that opens the modal and focuses on the input. When a user provides the image path in the input and submits the little form, it drops an image on the page. It's just like the plugin-tooltip except streamlined so the user doesn't have to click so much.

I think the mechanics of that kind of behavior are simple, but what makes it tricky in the current setup in milkdown is the fact that the same tooltip is used for lots of different behaviors. I think separate tooltips could be a good idea, because then you don't need such a complex tooltip interface (for setting the tooltip state, binding actions to be dispatched...). And if each modal were expressed as its own plugin, then plugins that call one of the modals (e.g. the menu and the slash command, both of which may want to show an image modal) can each require the necessary modal plugin actions and interfaces...

Reply:
Do you have interests to make it as a community plugin? I always love to hear about the idea like this. It can make users have more choices.

Reply:
Yes, I was thinking the same :) The UI is expressed in React right now but it'll be straightforward to make it vanilla. This weekend could be a good time for a MR...

Reply:
You can also create your own package if you don't want to bothered by the current milkdown toolchain and tech stack.

Reply:
haha, let's not throw the baby out with the bathwater! Milkdown is fantastic; it's the best I've seen. I just think there are a few small opportunities where we can make the user experience even better :)

Reply:
My concern is that this project is maintained by me and the designer. There's much more work to do if we decided to add a new type of component in the codebase.